homeassistant/components/sensor/awair.py

```diff
@@ -148,57 +162,53 @@ def state(self):
         return self._data.data[self._type]
 
     @property
-    def score(self):
-        """Return the aggregate Awair score."""
-        return self._data.data[ATTR_SCORE]
-
-    @property
-    def timestamp(self):
-        """Return the timestamp of the last sensor reading."""
-        return self._data.data[ATTR_TIMESTAMP]
+    def device_state_attributes(self):
+        """Return additional attributes."""
+        return self._data.attrs
 
     @property
     def unique_id(self):
         """Return the unique id of this entity."""
         return "{}_{}".format(self._uuid, self._type)
 
-    @property
-    def uuid(self):
-        """Return the API UUID of this entity."""
-        return self._uuid
-
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity."""
         return self._unit_of_measurement
 
+    @property
+    def should_poll(self):
+        """HA should poll us for state."""
+        return True
+
     async def async_update(self):
         """Get the latest data."""
         await self._data.async_update()
 
 
-class AwairData(Entity):
+class AwairData:
     """Get data from Awair API."""
 
-    def __init__(self, client, uuid):
+    def __init__(self, client, uuid, throttle):
         """Initialize the data object."""
         self._client = client
         self._uuid = uuid
         self.data = {}
+        self.attrs = {}
+        self.async_update = Throttle(throttle)(self._async_update)
 
-    @Throttle(THROTTLE)
-    async def async_update(self):
+    async def _async_update(self):
         """Get the data from Awair API."""
         resp = await self._client.air_data_latest(self._uuid)
+
         timestamp = datetime.strptime(resp[0][ATTR_TIMESTAMP], TIME_FORMAT)
+        self.attrs[ATTR_TIMESTAMP] = timestamp
```

Comment:
Is there a difference between this time and `last_updated` state attribute, where the latter attribute is when the state or state attributes last changed?

Reply:
It's the timestamp of the last time the device reported data to their servers. I think that's useful to have, because the API doesn't return connection state in any other way. I'll address the feedback though - it's clearly confusing, and I can make it less so. :)